src/inspector/inspector_socket_server.cpp

```diff
@@ -338,7 +338,8 @@ bool InspectorSocketServer::HasTargets() {
 
 bool InspectorSocketServer::Start() {
   state_ = ServerState::kRunning;
-  std::thread([this]() {
+  std::unique_lock<std::mutex> tcp_server_lock(mutex_tcp_server_);
+  std::thread([this, lock = std::move(tcp_server_lock)]() {

```

Comment:
>lock [](http://example.com/codeflow?start=21&length=4)

So this is locked even before thread has a chance to execute. Should this be scoped down to start of execution?

Reply:
If such extended lock is indeed an expected behavior, would you mind adding brief comment explaining it?

Reply:
It feels safer to lock before; don't know what would happen if we get through Start, then Stop, then the thread gets scheduled now touching deleted memory.

FWIW, in real-life this won't really matter; it's an issue in unit tests that quickly bring up and tear down runtimes (see 5610714 in Office).

Reply:
I see. If it's not too difficult, please make a comment in code about it.